src/Interpreters/executeQuery.cpp

```diff
@@ -796,10 +796,16 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
             catch (const Exception & e)
             {
                 if (e.code() == ErrorCodes::SYNTAX_ERROR)
-                    /// Don't print the original query text because it may contain sensitive data.
-                    throw Exception(ErrorCodes::LOGICAL_ERROR,
-                        "Inconsistent AST formatting: the query:\n{}\ncannot parse.",
-                        formatted1);
+                {
+                    if (context->getServerSettings().debug_ast_formatting_print_original_query)
```

Comment:
No need for a setting. It can be done by default in the debug build.

Reply:
OK. Done